lib/nx_deflate.c

```diff
@@ -630,9 +630,6 @@ int nx_deflateInit2_(z_streamp strm, int level, int method, int windowBits,
 		return Z_STREAM_ERROR;
 	}
 
-	/* only support level 6 here */
-	level = 6;
```

Comment:
You are enabling all levels here but level 0 is not working well at the moment and needs to be fixed.
I haven't tested the others extensively but they seem to work.